specification/webpubsub/resource-manager/Microsoft.SignalRService/preview/2024-01-01-preview/webpubsub.json

```diff
@@ -0,0 +1,4018 @@
+          "description": "Stop or start the resource.  Default to \"False\".\r\nWhen it's true, the data plane of the resource is shutdown.\r\nWhen it's false, the data plane of the resource is started.",
+          "default": "false",
+          "type": "string"
+        },
+        "socketIO": {
+          "$ref": "#/definitions/WebPubSubSocketIOSettings"
+        }
+      }
+    },
+    "WebPubSubRequestType": {
+      "description": "The incoming request type to the service",
+      "enum": [
+        "ClientConnection",
+        "ServerConnection",
+        "RESTAPI",
+        "Trace"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "WebPubSubRequestType",
+        "modelAsString": true
+      }
+    },
+    "WebPubSubResource": {
+      "description": "A class represent a resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/ResourceSku"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/WebPubSubProperties"
+        },
+        "kind": {
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "$ref": "#/definitions/ServiceKind"
+        },
+        "identity": {
+          "$ref": "#/definitions/ManagedIdentity"
+        }
+      }
+    },
+    "WebPubSubResourceList": {
+      "description": "Object that includes an array of resources and a possible link for next set.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List of the resources",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/WebPubSubResource"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "description": "The URL the client should use to fetch the next page (per server side paging).\r\nIt's null for now, added for future use.",
+          "type": "string"
+        }
+      }
+    },
+    "WebPubSubSkuTier": {
+      "description": "Optional tier of this particular SKU. 'Standard' or 'Free'. \r\n\r\n`Basic` is deprecated, use `Standard` instead.",
+      "enum": [
+        "Free",
+        "Basic",
+        "Standard",
+        "Premium"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "WebPubSubSkuTier",
+        "modelAsString": true
+      }
+    },
+    "WebPubSubSocketIOSettings": {
+      "description": "SocketIO settings for the resource",
+      "type": "object",
+      "properties": {
+        "serviceMode": {
+          "description": "The service mode of Web PubSub for Socket.IO. Values allowed: \r\n\"Default\": have your own backend Socket.IO server\r\n\"Serverless\": your application doesn't have a backend server",
+          "type": "string"
```

Comment:
seems `serviceMode` should be defined as an enum?

Reply:
To keep the case consistency in the user's request and the response, we use the string rather than enum to keep the case. This is also the way we did for the other similar properties.